### PR TITLE
Departmental clothing vendors (vendors only, no map edits)

### PR DESCRIPTION
**Pull request by E-MonaRhg**
## What Does This PR Do

~~Ruins the loadout economy~~
Adds clothing vendors, **not mapped in yet** but will be in later PRs.
Renames some clothes so they are more descriptive but more importantly so they don't break the vendors.

## Why It's Good For The Game

Departmental clothing lockers are a bit clunky, as in:

- you have to spend some time sorting through them to find what you need, and that something most likely isn't even there but hidden in loadouts
- some wardrobes, such as the medical one on the Cyberiad had all of its clothes mapped in, which is not a great practice. Vendors are way more modular and convenient.

Overall the locker wardrobes are a bit of a mess while clothing vendors are, in my opinion, way more convenient.

## Images of changes

Sprites ported from TG
All the vendors and their animations:
ruining_the_loadout_economy
_The gif loops, so it looks a bit weird here. Looks fine in-game._

## Changelog
:cl:
add: Added departmental clothing vendors.
tweak: Renames the medical scrubs, surgical caps, the black security jumpsuit and non-standard hard hats so their names are more descriptive and don't break the vendors
imageadd: Added icons for the clothing vendors
/:cl:

### CODE REVIEW

**Comment by S34NW** [comment]
YES

**Review by S34NW — APPROVED** [review]
Mapping wise, this PR is fine.
I have not looked at the code side of things yet.

**Comment by Sirryan2002** [comment]
Def thumbsup from me on this. Great sprites and it removes the need to clog up map queue for department clothing alterations.

**Comment by Wojz** [comment]
Genetics and robotics gets one but paramedic doesn't? They actually have unique stuff like EMT labcoat and the vest. :P

Also, the security bomber jacket's now an option for regular officers so it could be worth including.

File: _maps/map_files/cyberiad/cyberiad.dmm [hunk 1]
```diff
@@ -4468,13 +4468,13 @@
...
 /area/maintenance/fsmaint)
 "amf" = (
-/obj/structure/closet/secure_closet/security,
 /obj/structure/window/reinforced{
 	dir = 4
 	},
 /obj/item/radio/intercom{
 	pixel_y = 25
 	},
+/obj/machinery/vending/secdrobe,
```
Comment (SabreML) [inline _maps/map_files/cyberiad/cyberiad.dmm:4477]: Does sec really need *two* clothing vendors? I know fashion is the main focus of the department, but this is quite a lot of clothes for maybe a dozen people.
image
image

File: code/datums/supplypacks.dm [hunk 2]
```diff
@@ -228,6 +228,13 @@ GLOBAL_LIST_INIT(all_supply_groups, list(SUPPLY_EMERGENCY,SUPPLY_SECURITY,SUPPLY
 	contains = list(/obj/item/vending_refill/security)
 	containername = "SecTech supply crate"
 
+/datum/supply_packs/security/vending/clothingvendor
+	name = "Security Clothing Vendors Crate"
+	cost = 40
+	contains = list(/obj/item/vending_refill/secdrobe,
+					/obj/item/vending_refill/detdrobe)
+	containername = "security clothing vendor crate"
```
Comment (SabreML) [inline code/datums/supplypacks.dm:236]: The "Security Clothing Crate" and the "Security Clothing Vendors Crate" are a bit ambiguous when you search for them ingame, and I can imagine people might get confused by it:
image
You can check the contents of course, but it *might* be better if these resupply crates were renamed to "ThingDrobe Vendor Crate" or something instead. It's fine either way though.

**Comment by Darkmight9** [comment]
Might I suggest a possible idea that I've been thinking of? Janitor equipment vendor! Has both their uniforms and their gear in it!

**Comment by Spacemanspark** [comment]
This has duffelbags in at least the science/ medical vendor, right?

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -1778,3 +1778,326 @@
...
+	products = list(/obj/item/clothing/under/rank/medical/blue = 3,
+					/obj/item/clothing/under/rank/medical/green = 3,
+					/obj/item/clothing/under/rank/medical/purple = 3,
+					/obj/item/clothing/under/rank/medical = 3,
+					/obj/item/clothing/under/rank/medical/skirt = 3,
+					/obj/item/clothing/under/rank/nurse = 3,
+					/obj/item/clothing/under/medigown = 3,
+					/obj/item/clothing/head/surgery/blue = 3,
+					/obj/item/clothing/head/surgery/green = 3,
+					/obj/item/clothing/head/surgery/purple = 3,
```
[the remark this answers is not in the document]
Comment (E-MonaRhg) [reply code/game/machinery/vending.dm:1857]: Renamed the hard hats so they would will work properly now.
image

File: _maps/map_files/cyberiad/cyberiad.dmm (hunk 1 above)
Comment (E-MonaRhg) [reply _maps/map_files/cyberiad/cyberiad.dmm:4477]: Two were placed as to not destroy the symmetry of the equipment room as before this there was a corporate security clothing wardrobe and a normal security one, didn't want to keep just one wardrobe there when the vendor had it all anyway so I added two.

**Comment by E-MonaRhg** [comment]
>Genetics and robotics gets one but paramedic doesn't? They actually have unique stuff like EMT labcoat and the vest. :P

TG doesn't have a paramedic vendor and I'm no spriter so...
Can add them to the general MediDrobe though.

>Also, the security bomber jacket's now an option for regular officers so it could be worth including.

Added. Completely forgot that its not pod pilot exclusive now.

>Might I suggest a possible idea that I've been thinking of? Janitor equipment vendor! Has both their uniforms and their gear in it! Iirc TG has those too so could port it over as well

Was thinking about it but went the way of just having _clothing_ vendors, the TG janitor vendor has equipment too + janitors don't really get all that much clothes to pick from.
Could still add it though I suppose but I don't think its really all that needed.

>This has duffelbags in at least the science/ medical vendor, right?

Yes
image

And yes
image

File: code/datums/supplypacks.dm (hunk 2 above)
Comment (E-MonaRhg) [reply code/datums/supplypacks.dm:236]: I _myself_ don't really think its all that confusing, and I having a separate crate for every vendor would clutter the console a bit too much.

**Comment by Darkmight9** [comment]
> Was thinking about it but went the way of just having _clothing_ vendors, the TG janitor vendor has equipment too + janitors don't really get all that much clothes to pick from.
> Could still add it though I suppose but I don't think its really all that needed.

Yeah it should probably be done in a separate PR. I've been learning to try and avoid PR bloat

**Comment by Spacemanspark** [comment]
oh wait, shit
security vendor has the corporate jumpsuit/ cap, right?

**Comment by E-MonaRhg** [comment]
> security vendor has the corporate jumpsuit/ cap, right?

It has it all
image

**Review by Kyep — APPROVED** [review]

**Comment by E-MonaRhg** [comment]
Reverted map edits to let the map queue move faster and without clogging it with this PR which took three of the queue slots.

Separate PRs will be opened to add them to all three maps after this gets merged.

**Review by variableundefined — APPROVED** [review]
Good to go, maints requested the mapping change be spinned out

**Pull request merged**